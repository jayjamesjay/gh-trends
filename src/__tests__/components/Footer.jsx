import React from 'react';
import 'jest-styled-components';
import { render, screen } from '@testing-library/react';
import Footer from '../../components/Footer';

const text = 'Copyright Â© 2022 jayjamesjay. All Rights Reserved. Project under MIT licence.';

describe('<Footer />', () => {
  it(`renders with default style`, () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with correct copyright info', () => {
    render(<Footer />);
    expect(screen.getByRole('contentinfo').textContent).toEqual(text);
  });
});
