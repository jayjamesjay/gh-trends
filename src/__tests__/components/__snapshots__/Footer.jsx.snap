// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders with default style 1`] = `
<DocumentFragment>
  .c0 {
  margin: 1rem;
  padding: 0.5rem;
  text-align: center;
}

<footer
    class="c0"
  >
    <p>
      Copyright Â© 2025 jayjamesjay. All Rights Reserved. Project under MIT licence.
    </p>
  </footer>
</DocumentFragment>
`;
